packages/next/src/server/app-render/collect-segment-data.tsx

```diff
@@ -74,9 +124,17 @@ function collectSegmentDataImpl(
   serverConsumerManifest: any,
   segmentPath: Array<[string, Segment]>,
   segmentPathStr: string,
+  accessToken: string,
   segmentBufferMap: Map<string, Buffer>,
   collectedTasks: Array<Promise<void>>
-): void {
+): TreePrefetch {
+  // Metadata about the segment. Sent as part of the tree prefetch.
+  const slotMetadata: { [parallelRouteKey: string]: TreePrefetch } = {}
+
+  // Access tokens for the child segments. Sent as part of layout's data. Null
+  // if there are no children.
+  let childAccessTokens: { [parallelRouteKey: string]: string } | null = {}
```

Comment:
It seems like if there are no children, it'll be `{}`, at least in the current implementation
```suggestion
  // Access tokens for the child segments. Sent as part of layout's data. Null
  // if there are no children.
  let childAccessTokens: { [parallelRouteKey: string]: string } | null = null
```

Reply:
You mean for FlightRouterState? Yeah that's fine though since this is a different type. I prefer this pattern because if you need to know if it's a leaf node you can do a null check instead of checking the length of the keys

Reply:
What I meant is that I don't think `childAccessTokens` is ever null in the current approach like the comment says, since we initialize it to an empty object.

Reply:
it's initialized to null on this line? :D

```js
let childAccessTokens: { [parallelRouteKey: string]: string } | null = /* HERE */ null
```

The left side is the type

Reply:
oh lol nvm I was reading your suggestion

Reply:
yeah good catch

Reply:
Those inline object types always trip me up